Added links to required software for Windows. (#2692)
iterative/dvc.org: content/docs/user-guide/contributing/docs.md

Make sure you have a recent LTS version of [Node.js](https://nodejs.org/en/)
(`>=12.0.0`, `<=15.x`), and install [Yarn](https://yarnpkg.com/):

> > In Windows, you may need to install [Python], [Visual Studio Build Tools], and

Reviewer: Also why do you need Python to run `npm` and `yarn`?

Author: Not `npm` or `yarn` itself, but one of the (thousands of) packages require it. :)

Reviewer: To run the docs engine locally? I'd be surprised if that's the case. @rogermparent can you confirm? Thanks

And if it's true we should probably mention it in general, not just for Windows.

Author: I didn't take the note of which package, and I don't want to test by removing everything but yes, Python was required to be installed by a package.

Reviewer 2: [`node-gyp`](https://github.com/nodejs/node-gyp), which is used to build native addons like `sharp` (our image processor) depends on Python. No way around it, though I've never run into the situation where I don't have Python (probably most likely on Windows I'd guess).

Reviewer: Interesting.

@iesahin mind mentioning Python in general then?

Author: I don't think there is a Linux or Mac installation without Python. Should we really mention it?

Reviewer: Depends, what version of Python is needed? If it's 3.x then we should mention it as 2.x can still be the built-in one on MacOS and Linux I think.

Reviewer: Seems like we should mention for all platforms after all @iesahin ? Per https://github.com/iterative/dvc.org/pull/2692#issuecomment-929581152

Author: Added Python 3.6+ as a general requirement like Yarn.

> > [Windows SDK] additionally for dependencies.

Reviewer: ```suggestion
> [Windows SDK] first.
```

Reviewer: Why would the Windows SDK be needed though? Seems overkill, isn't it several GBs?

Author: I remember some package in the requirements asking for it. Some of these are compiled packages and probably need one or two functions from the SDK. Seems overkill, yes.

$ yarn
```

> In Windows, you may need to install [Python], [Visual Studio Build Tools], and
> [Windows SDK] beforehand for dependencies.

[windows sdk]:
  https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk/
[visual studio build tools]:
  https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019

> [python]: https://www.python.org/downloads/

Reviewer: Let's put this before all the code blocks? Otherwise it kind of interrupts the doc flow a bit.

Reviewer: TBH maybe we should recommend a POSIX env instead and link to https://dvc.org/doc/user-guide/running-dvc-on-windows for details.

Author: I was able to run the node on Windows. This is not about running DVC itself on Windows, this is about running dvc.org clone on Windows. 

I agree that using WSL / POSIX is better. But in order to contribute to documents on Windows (without all this WSL stuff first), one should be able to run node.js first. Otherwise, for example, running `yarn format-all` doesn't work for contributors.

Author: I moved it before the node installation command.

## Doc style guidelines (JavaScript and Markdown)

> Following rules are applied automatically by a pre-commit Git hook that is

Reviewer: ```suggestion
Some of the following rules are applied automatically by a pre-commit Git hook that is
```

> Some of the following rules are applied automatically by a pre-commit Git hook that is
> installed when `yarn` runs (see [dev env](#development-environment)).

Reviewer: Restyled didn't catch this for some reason... 😕 

```suggestion
Some of the following rules are applied automatically by a pre-commit Git hook
that is installed when `yarn` runs (see [dev env](#development-environment)).
```

Author: Somehow my VSCode prettier and yarn's prettier behave differently sometimes, in minor ways. There is also some discrepancy between Git's handling of CRLF and Prettier's config. It should be OK in my latest commit.

Reviewer: Maybe related: https://github.com/iterative/dvc.org/issues/2641

## Doc style guidelines (JavaScript and Markdown)

> Some of the following rules are applied automatically by a pre-commit Git hook

Reviewer: Thanks for the typo fix.

language such as "As you can see, clearly this is what happened, of course" etc.
which while good-intentioned, may scare readers off.

> We prefer human-friendly language rather than exact jargon as long as it's
> correct. Example: avoid Git jargon such as _revision_ or _reference_, preferring
> the more basic concept _commit_.

Reviewer: Out of scope to review the language in this guide but OK...
```suggestion
We prefer general, human-friendly language rather than exact jargon as long as it's
correct. Example: avoid Git jargon such as _revision_ or _reference_, preferring
the more basic terms _commit_ or _version_.
```

Start by writing the essence in simple terms, and complete it with
clarifications, edge cases, or other precisions in a separate iteration.

> We use **bold** text for emphasis and _italics_ for special terms.

Reviewer: This was OK.
